### PR TITLE
Testing To Do list: part 2

### PR DESCRIPTION
**Pull request by ntzwilly**
- Use jest framework for testing.
- Create tests for editTask and changeStatus functions.
- the "Clear all completed" function. (We did this yesterday)
- Use the mock storage object to mock the localStorage updates.
- Make sure you group your tests using the description() method.

### CODE REVIEW

File: src/__tests__/changeStatus.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    {
+      description: 'something to do 2',
+      completed: false,
+      index: 2,
+    },
+    {
+      description: 'something to do 3',
+      completed: true,
+      index: 3,
```
Comment (dasileker) [inline src/__tests__/changeStatus.test.js:19]: - [ ] The mock testing should be provided by jest [see here how](https://stackabuse.com/using-mocks-for-testing-in-javascript-with-jes/)

File: src/__mocks__/changeStatus.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  if (arr[index].completed) {
+    arr[index].completed = false;
+  } else {
+    arr[index].completed = true;
+  }
+  localStorageMock.setItem('data', arr);
+  return arr;
+};
+
+export default changeStatus;
```
Comment (dasileker) [inline src/__mocks__/changeStatus.js:13]: - [ ] As mentioned in the previous comment the mock testing it can be done using the jest documentation.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ntzwilly,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- The `AddTask.test.js` is failing, please add a valid test suite.

kkkk


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__mocks__/changeStatus.js (hunk 2 above)
Comment (chuanqizhu0820) [reply src/__mocks__/changeStatus.js:13]: Hi, thanks for reviewing. Could you be more specific about what we have done incorrectly here?

File: src/__tests__/changeStatus.test.js (hunk 1 above)
Comment (chuanqizhu0820) [reply src/__tests__/changeStatus.test.js:19]: Hi, the addTask.test.js run well on my computer. I can not figure what's wrong with our code. How should we do
Screen Shot 2021-09-30 at 3 39 40 PM
?

**Review by SpaYco — COMMENTED** [review]
Hello 👋,

_This is an additional review as a result of the second opinion request._

The tests run fine for me too, please skip that requirement.

I can also see that there's no _official_ way we can mock localStorage, so we can skip this as well, but I advise you to check a better way from [here](https://stackoverflow.com/questions/52883100/jest-mock-localstorage-methods)

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Chuanqi, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾 :shipit:.
Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**